lib/src/snapshot/snapshot_builder.dart

```diff
@@ -119,11 +120,18 @@ class _SnapshotBuilderState extends State<SnapshotBuilder> {
       RenderRepaintBoundary? boundary = _containerKey.currentContext
           ?.findRenderObject() as RenderRepaintBoundary?;
 
-      if (boundary == null || boundary.debugNeedsPaint) {
+      if (boundary == null) {
         completer.complete(null);
         return;
       }
 
+      if (kDebugMode) {
```

Comment:
Here you can merge two of these conditions to have "kDebugMode && boundary.debugNeedsPaint"